### ISSUE TITLE
Crash after updating to v18.6.0

### ISSUE THREAD

**Issue by evelynepinger** [GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk#129]
**Describe the bug**
After updating our recaptcha library to 18.6.0 (prev version was 18.3.0) we have started seeing a number of crashes from the recaptcha library.

**Integration Method**

Select the method used to integrate with reCAPTCHA Mobile.

* [x] SPM (iOS)
* [ ] Cocoapods (iOS)
* [ ] Direct Download (iOS)
* [ ] GMaven (Android)

**SDK Version (e.g. 18.0.1):**
18.6.0

**To Reproduce**

Could not reproduce.

Crash log:
```
          Crashed: recCQ
0  App                    0x117a68 specialized __RawDictionaryStorage.find<A>(_:) + 4373428840
1  App                    0x151e5a8 x3L3oBbL.lr4LWuAa(_:_:_:) + 4394427816
2  App                    0x1512024 closure #1 in jWf4TI6Y.swrofoep(_:_:) + 4394377252
3  App                    0x15139ec partial apply for closure #1 in jWf4TI6Y.swrofoep(_:_:) + 4394383852
4  App                    0x154fcd8 specialized closure #1 in static rcN96BVi.jpxIHp9r<A>(_:_:_:) + 4394630360
5  App                    0x1553974 partial apply for specialized closure #1 in static rcN96BVi.jpxIHp9r<A>(_:_:_:) + 4394645876
6  App                    0xbe248 thunk for @escaping @callee_guaranteed () -> () + 4373062216 (<compiler-generated>:4373062216)
7  libdispatch.dylib              0x12e5c <redacted> + 148
8  libdispatch.dylib              0x3dd4 <redacted> + 20
9  libdispatch.dylib              0x72d8 <redacted> + 600
10 libdispatch.dylib              0x68f4 <redacted> + 584
11 libdispatch.dylib              0x15894 <redacted> + 392
12 libdispatch.dylib              0x1609c <redacted> + 156
13 libsystem_pthread.dylib        0x48f8 _pthread_wqthread + 228
14 libsystem_pthread.dylib        0x10cc start_wqthread + 8
```
        

**Expected behavior**
No crashes.

**Screenshots**

**Xcode version for iOS (please complete the following information):**
 - Version [e.g. Xcode 13.4.1]

**Device (please complete the following information):**
 - Model: Multiple devices are affected.
 - OS: iOS 15, 16, 17, 18
 - Emulator/Simulator or real Device? 

**Additional context**

**Comment by walterjgsp** [comment]
Thanks for the report @evelynepinger, we have been investigating this issue as this is similar to what is being faced in the most recent report of https://github.com/GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk/issues/121. We were also not able to reproduce but we have some possible fixes in our newest release that should happen in a couple of weeks. I will keep you posted.

**Comment by walterjgsp** [comment]
Hi @evelynepinger. We released a new version 18.7.0-beta01 with what we expect to have fixed the issue you're facing. Please let me know if your problem is fixed using the new version.

**Comment by walterjgsp** [comment]
Hi @evelynepinger, we've just released iOS version 18.7.0. Please let me know if the issue you were experiencing is still present. We haven't been able to reproduce the problem, but we've addressed all identified bugs that might have been contributing to it.

**Comment by AshleyDeng** [comment]
Hey @walterjgsp , since the iOS version 18.7.0 is still in beta, what's your recommendation on upgrading the SDK version?

- Go straight to 18.7.0 beta
- Go with 18.6.0 (I hope the crash is not widespread)
- Go with 18.5.1, the stable release

**Comment by mcorner** [comment]
iOS and Android 18.7.0 is no longer in beta, but I think maybe the release notes are missing.  We will check.

**Comment by mcorner** [comment]
yeah, release notes are missing, will fix.  You can still use 18.7.0 (non beta).

**Comment by AshleyDeng** [comment]
Thanks @mcorner @walterjgsp!
We are using Carthage with a Direct Download of the SDK.

What other dependencies do you guys depend on? I think I also need to upgrade them on my end.

Thanks!

**Comment by AshleyDeng** [comment]
@walterjgsp @mcorner 

I was able to download and build the project with 18.6.0.
However, when I change the version and the URL to 18.7.0, the download is successful, but I am not able to build the project.

```
Unable to determine framework Swift version: No architectures found in dSYM.
```

This is the error that blocks the build.

**Comment by walterjgsp** [comment]
Hi @AshleyDeng, would you be able to provide a small, reproducible repository? This would allow us to run some tests. As we don't officially support Carthage, we lack the dedicated infrastructure to test it ourselves.

**Comment by erneestoc** [comment]
@walterjgsp thanks for your help! The issue @AshleyDeng refers to could work if the xcframework had a conventional `-Swift.h` header, or if Carthage checked other headers to see if it was generated by swift.

I submitted a PR to carthage, https://github.com/Carthage/Carthage/pull/3395

But if the xcframework could be re-generated with -Swift.h as header name that'd be great.

**Comment by walterjgsp** [comment]
Thanks for your response, @erneestoc. We investigated this with our build team and confirmed the issue lies with Carthage's handling of the `-Swift.h` header. Apparently, Xcode typically only uses that specific naming convention when another header file is present. Regenerating our framework just to address this would be difficult for us, so your fix within Carthage is the ideal solution. We really appreciate you implementing that change.

**Comment by AshleyDeng** [comment]
Hi @walterjgsp, could you further elaborate why the Recaptcha SDK was following the Apple's header file convention up until 18.6.0, but stop following it in 18.7.0?

cc: @erneestoc

**Comment by walterjgsp** [comment]
Hi @AshleyDeng, sure! In version 18.7.0, we moved the public API from ObjC to Swift. The build system now generates just one header file containing the Swift interface. Importantly, because it's the only header, we name it using the standard framework convention (e.g., `RecaptchaEnterprise.h`). This eliminates the need for separate headers and means Xcode no longer generates the `-Swift.h` file. 

Also something that I just notice is that the 18.6.0 also do not have the `-Swift.h` file but more headers. You can check the same if you use the direct download link here [18.6.0](https://dl.google.com/recaptchaenterprise/v18.6.0/RecaptchaEnterprise_iOS_xcframework/recaptcha-xcframework.xcframework.zip) and compare with [18.7.0](https://dl.google.com/recaptchaenterprise/v18.7.0/RecaptchaEnterprise_iOS_xcframework/recaptcha-xcframework.xcframework.zip). For 18.6.0 you will see `RecaptchaAction.h`, `RecaptchaEnterprise.h` and all public ObjC header files inside the `Headers` folder. So we never generated a `-Swift.h` file that was expected by Carthage. 

Feel free to reach out if you want to talk more about this.

**Comment by jludwiggreenaction** [comment]
Is there any hack/workaround for this while remaining on `v18.6.1`? It will be a big lift for us to upgrade due to changes in requirements of react native versions between `18.6` and `18.7`.

Unfortunately we did not see this issue before upgrading to `18.6` and now our only other option is to downgrade.
